.github/update-ghc.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+
+import os, sys, re
+import importlib
+import importlib.util
+import importlib.machinery
+import http.client
+import json
+from pathlib import PurePosixPath, Path
+from pprint import pprint
+from urllib.parse import quote
+from subprocess import Popen, PIPE, STDOUT, DEVNULL, check_output, run
+
+def import_path(path):
+    module_name = os.path.basename(path).replace('-', '_')
+    spec = importlib.util.spec_from_loader(
+        module_name,
+        importlib.machinery.SourceFileLoader(module_name, path)
+    )
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+    sys.modules[module_name] = module
+    return module
+
+
+GHC_GITLAB_SERVER="gitlab.haskell.org"
+
+def get_gitlab_tags_with_prefix(project_id, prefix):
+    connection = http.client.HTTPSConnection(GHC_GITLAB_SERVER)
+    encoded_project_id = encoded_project_id = quote(project_id, safe='')
+    endpoint = f"/api/v4/projects/{encoded_project_id}/repository/tags?search=^{prefix}&order-by=version"
+
+    connection.request("GET", endpoint)
+    response = connection.getresponse()
+
+    if response.status == 200:
+        tags_data = response.read().decode("utf-8")
+        tags = json.loads(tags_data)
+        return tags
+    else:
+        print(f"Error: {response.status} - {response.reason}")
+        return None
+
+project_id = "ghc/ghc"
+
+# matches GHC release versions
+version_re = re.compile("ghc-(?P<version>.*)-release")
+
+def parse_version(s):
+    return tuple(map(int, s.split(".")))
+
+def main():
+    SCRIPT_PATH = Path(__file__)
+
+    if len(sys.argv) != 2:
+        sys.exit(f"usage: {SCRIPT_PATH.name} GHC_MAJOR_MINOR")
+
+    bindist = import_path(str(SCRIPT_PATH.parent.parent.joinpath('haskell/private/ghc_bindist_generated.bzl')))
```

Comment:
An easier route could be to replace `haskell/private/ghc_bindist_generated.bzl` by a JSON file. The data is only loaded into the repository rule in `haskell/ghc_bindist.bzl`, so a combination of `repository_ctx.read` and `json.decode` could achieve the same thing. I think the data is only stored in a `.bzl` file for historic reasons (older versions of Bazel didn't expose JSON parsing in the Starlark API).